nixos/tests/nats.nix

```diff
@@ -0,0 +1,66 @@
+let
+
+  port = 4222;
+  username = "client";
+  password = "password";
+  topic = "foo.bar";
+
+in import ./make-test-python.nix ({ pkgs, lib, ... }: {
+  name = "nats";
+  meta = with pkgs.lib; { maintainers = with maintainers; [ c0deaddict ]; };
+
+  nodes = let
+    client = { pkgs, ... }: {
+      environment.systemPackages = with pkgs; [ natscli ];
+    };
+  in {
+    server = { pkgs, ... }: {
+      networking.firewall.allowedTCPPorts = [ port ];
+      services.nats = {
+        inherit port;
+        enable = true;
+        host = "0.0.0.0";
+        settings = {
+          authorization = {
+            users = [{
+              user = username;
+              inherit password;
```

Comment:
Oh no! :scream: 

What options does a sysadmin have for not storing passwords in the nix store? How sensitive is this password?

Reply:
Nats supports [bcrypted passwords](https://docs.nats.io/nats-server/configuration/securing_nats/auth_intro/username_password#bcrypted-passwords) in this field as well as plain text. How sensitive it is depends on the [permissions](https://docs.nats.io/nats-server/configuration/securing_nats/authorization) of the user. Besides basic username/password authentication Nats also supports [client TLS certificates](https://docs.nats.io/nats-server/configuration/securing_nats/auth_intro/tls_mutual_auth) or [NKeys which is based on Ed25519](https://docs.nats.io/nats-server/configuration/securing_nats/auth_intro/nkey_auth). I don't have any experience with these last two options, but it should be configurable with the generic settings JSON.

Reply:
Alright. Does this config file have includes? Can you specify multiple includes? Any ability at all to keep this out of the nix store?

If I understand correctly this isn't a big deal so these things I mention are "nice to haves".

Reply:
Apparently yes, it does support [includes](https://docs.nats.io/nats-server/configuration#include-directive). Unfortunately:

> Includes must use relative paths, and are relative to the main configuration (the one specified via the -c option):

Bit of a strange restriction, that is a deal breaker for including auth from outside of the nix store.

Reply:
Understood. If you don't have any ideas I suggest leaving as you have it now.

:+1: